frontend/src/domains/components/CategorySelector/CategorySelector.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState, useEffect } from 'react';
+import { useAppTheme } from '@/hooks/useAppTheme';
+import {
+  dropdownContainer,
+  dropdownButton,
+  dropdownList,
+  dropdownItem,
+  dropdownText,
+} from './CategorySelector.styles';
+import ArrowBottom from '@/components/icons/ArrowBottom';
+
+export interface SelectorOption {
+  value: string;
+  label: string;
+  disabled?: boolean;
+}
+
+export interface CategorySelectorProps {
+  width?: string | number;
+  options: SelectorOption[];
+  placeholder: string;
+  value?: string;
+  onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  id?: string;
+  name?: string;
+  disabled?: boolean;
+}
+
+export default function CategorySelector({
+  width = 'auto',
+  options,
+  placeholder,
+  value,
+  onChange,
+  id,
+  name,
+  disabled = false,
+}: CategorySelectorProps) {
+  const theme = useAppTheme();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const selectedOption = options.find((option) => option.value === value);
+  const displayText = selectedOption ? selectedOption.label : placeholder;
+
+  const handleToggle = () => {
+    if (!disabled) {
+      setIsOpen(!isOpen);
+    }
+  };
+
+  const handleSelect = (optionValue: string) => {
+    const fakeEvent = {
+      target: {
+        value: optionValue,
+        name: name || '',
+        id: id || '',
+      },
+    } as React.ChangeEvent<HTMLSelectElement>;
+
+    onChange?.(fakeEvent);
+    setIsOpen(false);
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const target = event.target as Element;
+      if (!target.closest('[data-category-selector]')) {
+        setIsOpen(false);
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener('click', handleClickOutside);
+      return () => document.removeEventListener('click', handleClickOutside);
+    }
+  }, [isOpen]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setIsOpen(false);
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener('keydown', handleKeyDown);
+      return () => document.removeEventListener('keydown', handleKeyDown);
+    }
+  }, [isOpen]);
```

Comment:
```suggestion
    const handleClickOutside = (event: MouseEvent) => {
      const target = event.target as Element;
      if (!target.closest('[data-category-selector]')) {
        setIsOpen(false);
      }
    };
    
    const handleKeyDown = (event: KeyboardEvent) => {
      if (event.key === 'Escape') {
        setIsOpen(false);
      }
    };

  useEffect(() => {
    if (isOpen) {
      document.addEventListener('keydown', handleKeyDown);
      return () => document.removeEventListener('keydown', handleKeyDown);
    }
  }, [isOpen]);
  
   useEffect(() => {
    if (isOpen) {
      document.addEventListener('click', handleClickOutside);
      return () => document.removeEventListener('click', handleClickOutside);
    }
  }, [isOpen]);
  
```

의존성에 영향받는게 아니라면 함수 선언은 useEffect 밖에서 해도 좋을 것 같아요
위의 useEffect도 하나의 useEffect로 묶을 수 있지 않을까 하고 생각이 들긴 하는데,
`return ()=>` 하는 부분에서 중복이 되는지를 잘 모르겠네요 ,,!

Reply:
이렇게 꼼꼼하게 해주시다니... 가능 여부 확인하고 수정해보겠습니다!